swagger.yaml

```diff
@@ -110,34 +244,79 @@ paths:
         404:
           description: "No dataset was found using the id provided"
         500:
-          description: "Failed to process the request due to an internal error"
+          $ref: '#/responses/InternalError'
   /datasets/{id}/editions/{edition}:
+    post:
+      tags:
+      - "Private user"
+      summary: "Create an edition of a dataset"
+      description: "Create an edition of a dataset"
+      parameters:
+      - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/new_edition'
+      responses:
+        201:
+          description: "A json object containing an edition"
+          schema:
+            $ref: '#/definitions/Edition'
+        400:
+          description: "Invalid request body"
+        401:
+          description: "Unauthorised to create/overwrite edition"
+        404:
+          description: "No dataset was found using the id provided"
+        500:
+          $ref: '#/responses/InternalError'
     get:
       tags:
-      - "Dataset"
+      - "Public"
       summary: "Get an edition of a dataset"
       description: "The edition contains a link to all versions"
       parameters:
-      - $ref: '#/parameters/id'
       - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/id'
       responses:
         200:
           description: "A json object containing an edition"
           schema:
             $ref: '#/definitions/Edition'
         404:
-          description: "No dataset edition was found using the id and edition provided"
+          description: "No edition of a dataset was found using the id and edition provided"
         500:
-          description: "Failed to process the request due to an internal error"
+          $ref: '#/responses/InternalError'
   /datasets/{id}/editions/{edition}/versions:
-    get:
+    post:
       tags:
-      - "Dataset"
-      summary: "Get a list of all versions from a dataset edition"
-      description: "Get a list of all versions from a dataset edition"
+      - "Private user"
+      summary: "Create a version for an edition of a dataset"
+      description: "Create a version for an edition of a dataset"
       parameters:
       - $ref: '#/parameters/id'
+      - $ref: '#/parameters/new_version'
+      responses:
+        201:
+          description: "A json object containing a version"
+          schema:
+            $ref: '#/definitions/NewVersionResponse'
+        400:
+          description: "Invalid request body"
+        401:
+          description: "Unauthorised to create/overwrite version of dataset"
+        403:
```

Comment:
There is not a scenario to return `Forbidden` status, this is because the version number is calculated within the API on POST request. If a private user sends a new version document, the api will check all versions against the specified edition and dataset and append by 1. Creating a wholly new document and never over-riding an existing document - its all in the endpoint `/datasets/{id}/editions/{edition}/versions` <- there is no version parameter in path or body